examples/webgpu_caustics.html

```diff
@@ -0,0 +1,247 @@
+			let stats;
+			let gltf;
+
+			init();
+
+			async function init() {
+
+				camera = new THREE.PerspectiveCamera( 25, window.innerWidth / window.innerHeight, 0.025, 5 );
+				camera.position.set( - 0.5, 0.35, 0.2 );
+
+				scene = new THREE.Scene();
+
+				// light
+
+				const spotLight = new THREE.SpotLight( 0xffffff, 1 );
+				spotLight.position.set( .2, .3, .2 );
+				spotLight.castShadow = true;
+				spotLight.angle = Math.PI / 6;
+				spotLight.penumbra = 1;
+				spotLight.decay = 2;
+				spotLight.distance = 0;
+				spotLight.shadow.mapType = THREE.HalfFloatType; // For HDR Caustics
+				spotLight.shadow.mapSize.width = 1024;
+				spotLight.shadow.mapSize.height = 1024;
+				spotLight.shadow.camera.near = .1;
+				spotLight.shadow.camera.far = 1;
+				spotLight.shadow.bias = - .003;
+				spotLight.shadow.intensity = .95;
+				scene.add( spotLight );
+
+				// model / textures
+
+				const dracoLoader = new DRACOLoader();
+				dracoLoader.setDecoderPath( 'jsm/libs/draco/' );
+				dracoLoader.setDecoderConfig( { type: 'js' } );
+
+				gltf = ( await new GLTFLoader().setDRACOLoader( dracoLoader ).loadAsync( './models/gltf/duck.glb' ) ).scene;
+				gltf.scale.setScalar( .5 );
+				scene.add( gltf );
+
+				const causticMap = new THREE.TextureLoader().load( './textures/opengameart/Caustic_Free.jpg' );
+				causticMap.wrapS = causticMap.wrapT = THREE.RepeatWrapping;
+				causticMap.colorSpace = THREE.SRGBColorSpace;
+
+				// objects / material
+
+				const duck = gltf.children[ 0 ];
+				duck.material = new THREE.MeshPhysicalNodeMaterial();
+				duck.material.side = THREE.DoubleSide;
+				duck.material.transparent = true;
+				duck.material.color = new THREE.Color( 0xFFD700 );
+				duck.material.transmission = 1;
+				duck.material.thickness = .25;
+				duck.material.ior = 1.5;
+				duck.material.metalness = 0;
+				duck.material.roughness = .1;
+				duck.castShadow = true;
+
+				// tsl shader
+
+				const causticOcclusion = uniform( 20 );
+
+				duck.material.castShadowPositionNode = Fn( () => {
+
+					// optional: add some distortion to the geometry shadow position if needed
+
+					return positionLocal;
+
+				} )();
+
+				duck.material.castShadowNode = Fn( () => {
+
+					const refractionVector = refract( positionViewDirection.negate(), normalView, div( 1.0, duck.material.ior ) ).normalize();
+					const viewZ = normalView.z.pow( causticOcclusion );
+
+					const textureUV = refractionVector.xy.mul( .6 );
+
+					const causticColor = uniform( duck.material.color );
+					const chromaticAberrationOffset = normalView.z.pow( - .9 ).mul( .004 );
+
+					const causticProjection = vec3(
+						texture( causticMap, textureUV.add( vec2( chromaticAberrationOffset.x.negate(), 0 ) ) ).r,
+						texture( causticMap, textureUV.add( vec2( 0, chromaticAberrationOffset.y.negate() ) ) ).g,
+						texture( causticMap, textureUV.add( vec2( chromaticAberrationOffset.x, chromaticAberrationOffset.y ) ) ).b
+					);
```

Comment:
@sunag It seems like there might be a typo here. `chromaticAberrationOffset` is a float, but is being accessed with `.x` and `.y`?

Reply:
Fixing, thank you :)